cdc/sorter/db/buffer.go

```diff
@@ -74,16 +76,27 @@ func (b *outputBuffer) shiftResolvedEvents(index int) {
 	}
 }
 
-// appendResolvedEvent appends resolved events to the buffer.
-func (b *outputBuffer) appendResolvedEvent(event *model.PolymorphicEvent) {
+// tryAppendResolvedEvent try to append resolved events to the buffer.
+// Return false if the buffer is full and append fails.
+func (b *outputBuffer) tryAppendResolvedEvent(event *model.PolymorphicEvent) bool {
 	if len(b.resolvedEvents) > 0 {
 		if b.resolvedEvents[0].CRTs != event.CRTs {
 			log.Panic("commit ts must be equal",
 				zap.Uint64("newCommitTs", event.CRTs),
 				zap.Uint64("commitTs", b.resolvedEvents[0].CRTs))
 		}
+	} else if len(b.resolvedEvents) == 0 {
```

Comment:
Can check `if len(b.resolvedEvents) == 0` first so that the `if len(b.resolvedEvents) > 0` can be avoids.